src/clue/graph/graph.ts

```diff
@@ -116,8 +116,12 @@ export class GraphEdge extends AttributeContainer {
   }
 
   private init() {
-    this.source.outgoing.push(this);
-    this.target.incoming.push(this);
+    if (this.source) {
+      this.source.outgoing.push(this);
+    }
+    if (this.target) {
+      this.target.incoming.push(this);
+    }
```

Comment:
This is already covered in the constructor (line 106). But `init()` is also called in `restore()` in line 155. Probably the return values of `nodes()` for source and target is `null`.